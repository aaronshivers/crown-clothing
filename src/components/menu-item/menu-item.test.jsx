import React from 'react';
import { shallow } from 'enzyme';
import MenuItem from './menu-item.component';

describe('MenuItem', () => {
  const props = {
    title: 'Title',
    imageUrl: 'ImageUrl',
  };

  const menuItem = shallow(
    <MenuItem title={props.title} imageUrl={props.imageUrl} />,
  );

  it('should render correctly', () => {
    expect(menuItem).toMatchSnapshot();
  });

  it('should have a class of `menu-item`', () => {
    expect(menuItem.hasClass('menu-item')).toBeTruthy();
  });

  it('should contain a `background-image` element', () => {
    expect(menuItem.find('.background-image')).toHaveLength(1);
  });

  it('should contain a `content` element', () => {
    expect(menuItem.find('.content')).toHaveLength(1);
  });

  it('should contain a `title` element', () => {
    expect(menuItem.find('.title')).toHaveLength(1);
  });

  it('should contain a `subtitle` element', () => {
    expect(menuItem.find('.subtitle')).toHaveLength(1);
  });
});
