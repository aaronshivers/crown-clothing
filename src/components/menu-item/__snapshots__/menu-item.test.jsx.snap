// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should render correctly 1`] = `
<div
  className="menu-item"
>
  <div
    className="content"
  >
    <h1
      className="title"
    >
      Title
    </h1>
    <span
      className="subtitle"
    >
      Subtitle
    </span>
  </div>
</div>
`;
